File: day-1/2.ts
export {};

const input = await Deno.readTextFile("./input.txt");

const elves = input.split("\n\n").map((elf) => {
  return elf.split("\n").map((elf) => parseInt(elf));
});

const elvesList: number[] = [];

for (let i = 0; i < elves.length; i++) {
  const total = elves[i].reduce((a, b) => a + b, 0);

  elvesList.push(total);
}

// find sum of top 3 in elvesList
const highest = elvesList
  .sort((a, b) => b - a)
  .slice(0, 3)
  .reduce((a, b) => a + b, 0);

console.log(highest);

File: day-1/2.ex
File.read!("./input.txt")
|> String.split("\n\n")
|> Enum.map(fn elf ->
  elf
  |> String.split("\n")
  |> Enum.map(fn elf -> elf |> String.to_integer() end)
  |> Enum.reduce(fn a, b -> a + b end)
end)
|> Enum.sort(:desc)
|> Enum.slice(0..2)
|> Enum.sum()
|> IO.inspect()
